application/bsdd_utils.py

```diff
@@ -0,0 +1,57 @@
+import database
+import json
+
+def get_hierarchical_bsdd(id):
+    with database.Session() as session:
+        model = session.query(database.model).filter(
+            database.model.code == id).all()[0]
+
+        hierarchical_bsdd_results = {}
+        if model.status_bsdd != 'n':
+            for bsdd_result in model.tasks[2].results:
```

Comment:
We shouldn't rely on index 2 here. There ordering may not even be consistent now, and could easily change in the future.